src/Settings/EquationSystem.cpp

```diff
@@ -136,8 +136,11 @@ void SimulationGenerator::ConstructEquations(
         CollisionQuantityHandler *cqh = ConstructCollisionQuantityHandler(re_type, runawayGrid, unknowns, ionHandler, s);
         eqsys->SetRunawayCollisionHandler(cqh);
     }
-    RunawayFluid *REF = ConstructRunawayFluid(fluidGrid,unknowns,ionHandler,re_type,s);
+    AnalyticDistributionRE *distRE = nullptr;
+    AnalyticDistributionHottail *distHT = nullptr;
+    RunawayFluid *REF = ConstructRunawayFluid(fluidGrid,unknowns,ionHandler,distRE,distHT,re_type,s);
     eqsys->SetREFluid(REF);
+    eqsys->SetAnalyticDists(distRE, distHT);
```

Comment:
I think that in some other places I have instead passed ``eqsys`` to the ``ConstructXXX()`` method so that it can call things like ``eqsys->SetAnalyticDist(...)`` instead of having these lines here, and passing in variables which are really just additional return parameters.

Reply:
Ah, that sounds like a good idea. I much prefer that solution!